src/v/storage/batch_cache.h

```diff
@@ -308,6 +300,18 @@ class batch_cache {
      */
     bool is_memory_reclaiming() const { return _is_reclaiming; }
 
+    /**
+     * @brief The estimated size of the cache in bytes.
+     *
+     * This is a slight underestimate of the true size of the memory used by the
+     * cache since it only counts the size of each underlying iobuf (arena) but
+     * not other overhead such as the iobuf struct itself, index structures,
+     * etc.
+     *
+     * @return size_t estimate of the in-memory size of the cache
+     */
+    size_t size_bytes() const { return _size_bytes; }
```

Comment:
Used by the reclaimer reporter and seems fine to expose to the public in general.